### DISCUSSION TITLE
Angluar Three Cannon Physics Examples

### DISCUSSION THREAD

**Discussion by IRobot1** [nartc/angular-three#105] [Show and tell]
If you need examples of using Angular Three Cannon physics, see this [demo](https://ng3cannon.z9.web.core.windows.net/)

![image](https://user-images.githubusercontent.com/25032599/165147983-3bf00429-c6fa-4067-900d-d8fc917462bf.png)

If you want to reference a specific example, add ?example=<name> to URL above.  For example, [cable](https://ng3cannon.z9.web.core.windows.net?example=cable).

**Comment by IRobot1** [comment]
Added example of [rolling ball](https://ng3cannon.z9.web.core.windows.net/?example=rolling)

![image](https://user-images.githubusercontent.com/25032599/165148575-5f4de86d-0060-484f-8bcd-a0ead747d8bd.png)

WASD to move.  Space to jump.

**Reply by nartc** [reply]
Haha this is so cool. Thank you @IRobot1 for all these examples (I think we should add your demo as a link from Cannon API documentations). 

BTW, WASD is a little wonky on my machine. Arrow keys work great!

**Comment by IRobot1** [comment]
Finally got the [cloth example](https://ng3cannon.z9.web.core.windows.net/?example=cloth) working.

![image](https://user-images.githubusercontent.com/25032599/167257053-2f955c75-3cd8-4755-9581-d59af4b2e853.png)

**Comment by IRobot1** [comment]
[Raycast Vehicle Physics Example](https://ng3cannon.z9.web.core.windows.net/?example=raycast_vehicle) now working

![image](https://user-images.githubusercontent.com/25032599/168687603-860ac63c-d8fb-4318-93fa-d85301c60b14.png)

**Comment by IRobot1** [comment]
Added [physics directives for Box, Sphere, Cylinder, Plane and Compound objects](https://github.com/IRobot1/ng3-cannon-template/tree/master/src/app/directives).  This allows declaratively adding physics to existing NGT meshes without having to write Typescript.  

Before

```html
<ngt-mesh [ref]="cubeProps.ref">
  <ngt-box-geometry></ngt-box-geometry>
  <ngt-mesh-standard-material wireframe color='red'></ngt-mesh-standard-material>
</ngt-mesh>
```

```ts
import { NgtPhysicBody } from "@angular-three/cannon";
@Component({
  selector:'some-component,
  template: '',
  providers: [NgtPhysicBody],
})
export class ThreeJSExample {
  cubeProps = this.physicBody.useBox(() => ({
    mass: 1,
    angularVelocity: [0, 10, 0],
    angularDamping: 0.5,
  }));

  constructor(private physicBody: NgtPhysicBody) { }
}
```

After
```html
<ngt-mesh physicsBox boxMass="1"
          [boxAngularVelocity]="[0, 10, 0]"
          boxAngularDamping="0.5">
  <ngt-box-geometry></ngt-box-geometry>
  <ngt-mesh-standard-material wireframe color='red'></ngt-mesh-standard-material>
</ngt-mesh>
```

```ts
@Component({
  selector:'some-component',
  template: '',
})
export class ThreeJSExample {
}
```

By default, physics objects are static (0 mass).  So very convenient to quickly define a wall or floor.  

All directives allow physics to be enabled/disabled at runtime too
```html
<ngt-mesh [physicsBox]="enabled" boxMass="1"
          [boxAngularVelocity]="[0, 10, 0]"
          boxAngularDamping="0.5">
  <ngt-box-geometry></ngt-box-geometry>
  <ngt-mesh-standard-material wireframe color='red'></ngt-mesh-standard-material>
</ngt-mesh>
```
